cli/exchange/business.go

```diff
@@ -0,0 +1,146 @@
+	}
+
+	//add/overwrite business policy file
+	httpCode := cliutils.ExchangePutPost(http.MethodPost, cliutils.GetExchangeUrl(), "orgs/"+org+"/business/policies"+cliutils.AddSlash(policy), cliutils.OrgAndCreds(org, credToUse), []int{201, 403}, policyFile)
+	if httpCode == 403 {
+		cliutils.ExchangePutPost(http.MethodPut, cliutils.GetExchangeUrl(), "orgs/"+org+"/business/policies"+cliutils.AddSlash(policy), cliutils.OrgAndCreds(org, credToUse), []int{201, 404}, policyFile)
+		fmt.Println("Business policy: " + org + "/" + policy + " updated in the Horizon Exchange")
+	} else {
+		fmt.Println("Business policy: " + org + "/" + policy + " added in the Horizon Exchange")
+	}
+}
+
+//BusinessUpdatePolicy will replace a single attribute of a business policy in the Horizon Exchange
+func BusinessUpdatePolicy(org string, credToUse string, policy string, attribute string, valueFilePath string) {
+	cliutils.SetWhetherUsingApiKey(credToUse)
+	org, credToUse = cliutils.TrimOrg(org, credToUse)
+
+	//verify that the policy exists
+	var exchangePolicy exchange.GetBusinessPolicyResponse
+	httpCode := cliutils.ExchangeGet(cliutils.GetExchangeUrl(), "orgs/"+org+"/business/policies"+cliutils.AddSlash(policy), cliutils.OrgAndCreds(org, credToUse), []int{200, 404}, &exchangePolicy)
+	if httpCode == 404 {
+		cliutils.Fatal(cliutils.NOT_FOUND, "Policy %s not found in org %s", policy, org)
+	}
+
+	//Read in patch and send to the exchange if format is correct
+	newBytes := cliconfig.ReadJsonFileWithLocalConfig(valueFilePath)
+	switch attribute {
+	case "service":
+		var newValue businesspolicy.ServiceRef
+		patch := make(map[string]businesspolicy.ServiceRef)
+		err := json.Unmarshal(newBytes, &newValue)
+		if err != nil {
+			cliutils.Fatal(cliutils.JSON_PARSING_ERROR, "failed to unmarshal json input file %s: %v", valueFilePath, err)
+		}
+		patch[attribute] = newValue
+		cliutils.ExchangePutPost(http.MethodPatch, cliutils.GetExchangeUrl(), "orgs/"+org+"/business/policies"+cliutils.AddSlash(policy), cliutils.OrgAndCreds(org, credToUse), []int{201}, patch)
+		fmt.Println("Policy " + org + "/" + policy + " updated in the Horizon Exchange")
+	case "properties":
+		var newValue externalpolicy.PropertyList
+		patch := make(map[string]externalpolicy.PropertyList)
+		err := json.Unmarshal(newBytes, &newValue)
+		if err != nil {
+			cliutils.Fatal(cliutils.JSON_PARSING_ERROR, "failed to unmarshal json input file %s: %v", valueFilePath, err)
+		}
+		err = newValue.Validate()
+		if err != nil {
+			cliutils.Fatal(cliutils.CLI_INPUT_ERROR, "Invalid format for properties")
+		}
+		patch[attribute] = newValue
+		cliutils.ExchangePutPost(http.MethodPatch, cliutils.GetExchangeUrl(), "orgs/"+org+"/business/policies"+cliutils.AddSlash(policy), cliutils.OrgAndCreds(org, credToUse), []int{201}, patch)
+		fmt.Println("Policy " + org + "/" + policy + " updated in the Horizon Exchange")
```

Comment:
The patch should take more than one attributes at a time. You do not have to break them up. This is my understanding. Can you confirm it with bp?

Reply:
Currently it only accept the first attribute in the input. Will need to get back to it once supporting multiple changes.